src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -114,15 +132,33 @@ ConvSolution InitInvokerFactoryNHWC(const ExecutionContext&,
     }
 
     ConvSolution result;
-    result.invoker_factory =
-        [ck_args     = CKArgsType{problem},
-         sh_conv_ptr = std::shared_ptr{std::move(*ptr_iter)}](const std::vector<Kernel>&) mutable {
-            return [ck_args = std::move(ck_args), sh_conv_ptr = std::move(sh_conv_ptr)](
-                       const Handle& handle, const AnyInvokeParams& primitive_parameters) {
-                const auto& data_ctx = primitive_parameters.CastTo<CastType>();
-                auto argument_ptr    = ck_args.MakeArgPtr(sh_conv_ptr, data_ctx.tensors);
-                auto invoker_ptr     = sh_conv_ptr->MakeInvokerPointer();
+    result.invoker_factory = [ck_args     = CKArgsType{problem},
+                              sh_conv_ptr = std::shared_ptr{std::move(*ptr_iter)}](
+                                 const std::vector<Kernel>&) mutable {
+        return [ck_args = std::move(ck_args), sh_conv_ptr = std::move(sh_conv_ptr)](
+                   const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            const auto& data_ctx = primitive_parameters.CastTo<CastType>();
+            auto argument_ptr    = ck_args.MakeArgPtr(sh_conv_ptr, data_ctx.tensors);
+            auto invoker_ptr     = sh_conv_ptr->MakeInvokerPointer();
+
+            if constexpr(std::is_same<CastType, miopen::conv::WrWInvokeParams>::value)
+            {
+                auto zero           = 0.0f;
+                const auto& tensors = data_ctx.tensors;
+                SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
 
+                HipEventPtr start = nullptr;
+                HipEventPtr stop  = nullptr;
+                if(handle.IsProfilingEnabled())
+                {
+                    ProfilingRecordStart(handle, start, stop);
+                }
+                invoker_ptr->Run(argument_ptr.get(), {handle.GetStream(), false});
+                if(handle.IsProfilingEnabled())
+                    ProfilingRecordStop(handle, start, stop);
+            }
+            else
+            {
```

Comment:
[Warning] At line 164 we are still getting elapsed time from CK, which, are you've said, leads to bugs (or the problem in CK impacts only WrW kernels?) Anyway, resolving my previous review comment would fix this.

Reply:
> [Warning] At line 164 we are still getting elapsed time from CK, which, are you've said, leads to bugs (or the problem in CK impacts only WrW kernels?) Anyway, resolving my previous review comment would fix this.

Yes. Only wrw kernels has the issue. I will fix and restructure the code base your comments.